src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -580,6 +595,41 @@ export const AutomatedAnalysisCard: React.FC<AutomatedAnalysisCardProps> = (prop
     dispatch(automatedAnalysisDeleteAllFiltersIntent(targetConnectURL));
   }, [dispatch, targetConnectURL]);
 
+  const icon = React.useCallback((score: number): JSX.Element => {
+    return score == AutomatedAnalysisScore.NA_SCORE ? (
+      <span className={css('pf-m-grey', 'pf-c-label__icon')}>
+        <InfoCircleIcon />
+      </span>
+    ) : score < AutomatedAnalysisScore.ORANGE_SCORE_THRESHOLD ? (
+      <span className={css('pf-m-green', 'pf-c-label__icon')}>
+        <CheckCircleIcon />
+      </span>
+    ) : score < AutomatedAnalysisScore.RED_SCORE_THRESHOLD ? (
+      <span className={css('pf-m-orange', 'pf-c-label__icon')}>
+        <ExclamationTriangleIcon />
+      </span>
+    ) : (
+      <span className={css('pf-m-red', 'pf-c-label__icon')}>
+        <ExclamationCircleIcon />
+      </span>
+    );
+  }, []);
+
+  const getSortParams = React.useCallback(
+    (columnIndex: number): ThProps['sort'] => ({
+      sortBy: {
```

Comment:
I think this type is `ISortBy` from `@patternfly/react-table`. 

So, we can just keep a state like below instead of 2 separate state for each field (i.e. activeSortIndex and activeSortDirection`.

https://github.com/cryostatio/cryostat-web/blob/23c440b242c30e4d23a40d960374e062e0eae72f/src/app/Agent/AgentProbeTemplates.tsx#L100